### DISCUSSION TITLE
Get automations with rest api

### DISCUSSION THREAD

**Discussion by PhantomCarrot** [Budibase/budibase#13697] [Help] [answered]
Hello!

My need is to display in a screen the list of automations of type cron to my users.
Direct display seems not possible, so i wanted to populate a table from a automation task and do things the classic way

I didn't find any public api for that but i saw that some internal api are undocumented (https://github.com/Budibase/budibase/discussions/1915)
I'm wondering if i can do the same for https://xxx.budibase.app/api/automations ? (it return enough infos) 

I cannot make it works by playing with it, anyone can help or tell me if it's not possible ?

Regards

**Comment by PhantomCarrot** [comment]
No one have ever faced this need ?

**Reply by ConorWebb96** [reply, marked as answer]
Hey @PhantomCarrot,

The Automations API available to you is an internal one, due to this it is not documented for public use.

You are welcome to use this solution at your own risk. Please be aware that if it stops working, we will not be able to provide support since it is not intended for external use.

If you are happy enough with this, heres how you can get all listed automations related to a specific app.

Similar to how you would query rows via our API. You will need an API key plus the ID of the app you are working with.

1. Get your API - in the app page of your builder click your profile -> `View api key` -> copy
2. Get your appID - can be found in the url of the builder you are currently on `app_dev_`
3. Remove the `_dev` as it will need to target your published app.
4. Hit the automations api route `[hostingURLHERE]/api/automations`
5. Create a new REST query with the above URL
6. Add headers e.g. `x-budibase-app-id` and `x-budibase-api-key` paste relevant values into these
7. Save and send.
8. This should provide you with a list of all automations related to your app.
![Screenshot 2024-05-21 at 08 37 29](https://github.com/Budibase/budibase/assets/126772285/da6d822d-77b7-4537-9069-8bfd417fbc3f)

I hope this helps.
